### PR TITLE
RavenDB-16955: add traffic watch logging tool

### PR DESCRIPTION
**Pull request by haludi**
https://github.com/ravendb/ravendb/pull/12465/files

### CODE REVIEW

File: tools/Raven.Debug/CommandLineApp.cs [hunk 1]
```diff
@@ -215,5 +220,91 @@ public static int Run(string[] args)
...
+
+        private static void ConfigureTrafficWatchCommand()
+        {
+            _app.Command("traffic", cmd =>
+            {
+                cmd.ExtendedHelpText = cmd.Description = "Log traffic watch entries.";
+                cmd.HelpOption(HelpOptionString);
+
+                var urlArg = cmd.Option("--url", "Server URL.", CommandOptionType.SingleValue);
+                var pathArg = cmd.Option("--path", "Directory path to save the traffic watch log.", CommandOptionType.SingleValue);
```
Comment (ppekrol) [inline tools/Raven.Debug/CommandLineApp.cs]: `--output-path`

File: tools/Raven.Debug/CommandLineApp.cs [hunk 2]
```diff
@@ -215,5 +220,91 @@ public static int Run(string[] args)
...
+                var databaseArg = cmd.Option("--database", "The database name.", CommandOptionType.SingleOrNoValue);
+                var typesArg = cmd.Option("--types", "Path to save the traffic watch logs.", CommandOptionType.MultipleValue); //TODO
+                var certArg = cmd.Option("--cert", "Path to pfx certificate file.", CommandOptionType.SingleOrNoValue);
```
Comment (ppekrol) [inline tools/Raven.Debug/CommandLineApp.cs]: `--certificate-path`

File: tools/Raven.Debug/CommandLineApp.cs [hunk 3]
```diff
@@ -215,5 +220,91 @@ public static int Run(string[] args)
...
+                var certPassArg = cmd.Option("--certpass", "Certificate password.", CommandOptionType.SingleOrNoValue);
```
Comment (ppekrol) [inline tools/Raven.Debug/CommandLineApp.cs]: `--certificate-password`

File: tools/Raven.Debug/CommandLineApp.cs [hunk 4]
```diff
@@ -215,5 +220,91 @@ public static int Run(string[] args)
...
+                var verboseArg = cmd.Option("-v", "Verbose to console.", CommandOptionType.NoValue);
```
Comment (ppekrol) [inline tools/Raven.Debug/CommandLineApp.cs]: `--verbose`

File: tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+        }
+
+        public async Task ConnectInternal()
+        {
+            try
+            {
+                var urlBuilder = new StringBuilder(_url).Append("/admin/traffic-watch");
+
+                //TODO Should fix server side
+                // if (string.IsNullOrWhiteSpace(_database) == false)
```
Comment (ppekrol) [inline tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs]: ?

File: tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+                                        {
+                                            if (json.TryGet("Type", out TrafficWatchChangeType type) == false)
+                                                continue;
+                                            if (_changeTypes.Contains(type) == false)
+                                                    continue;
+                                        }
+
+                                        if (_database != null)
+                                        {
+                                            if(json.TryGet("DatabaseName", out LazyStringValue databaseName) == false ||
```
Comment (ppekrol) [inline tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs:189]: can we use nameof?

**Review by ppekrol — CHANGES_REQUESTED** [review]
PR does not compile

File: tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs (hunk 6 above)
Comment (haludi) [reply tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs:189]: `TrafficWatchChangeBase` is internal and I don't want to change that.

File: tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs (hunk 5 above)
Comment (haludi) [reply tools/Raven.Debug/LogTrafficWatch/LogTrafficWatch.cs]: https://github.com/ravendb/ravendb/pull/12492

File: tools/Raven.Debug/CommandLineApp.cs [hunk 7]
```diff
@@ -215,5 +220,91 @@ public static int Run(string[] args)
...
+        {
+            _app.Command("traffic", cmd =>
+            {
+                cmd.ExtendedHelpText = cmd.Description = "Log traffic watch entries.";
+                cmd.HelpOption(HelpOptionString);
+
+                var urlArg = cmd.Option("--url", "Server URL.", CommandOptionType.SingleValue);
+                var pathArg = cmd.Option("--output-path", "Directory path to save the traffic watch log.", CommandOptionType.SingleValue);
+                var databaseArg = cmd.Option("--database", "The database name.", CommandOptionType.SingleOrNoValue);
+                var typesArg = cmd.Option("--types", "Path to save the traffic watch logs.", CommandOptionType.MultipleValue);
```
Comment (ppekrol) [inline tools/Raven.Debug/CommandLineApp.cs]: Wrong description

**Pull request merged**